pkg/operation/botanist/operatingsystemconfig.go

```diff
@@ -395,6 +395,29 @@ func (b *Botanist) generateCloudConfigExecutionChart() (*chartrenderer.RenderedC
 			w["command"] = *cmd
 		}
 
+		if worker.KubeletDataVolumeName != "" && worker.DataVolumes != nil {
+			for _, dataVolume := range worker.DataVolumes {
+				volName := dataVolume.Name
+				if *volName == worker.KubeletDataVolumeName {
+					size, err := resource.ParseQuantity(dataVolume.Size)
+					if err != nil {
+						return nil, err
+					}
+					sizeInBytes, ok := size.AsInt64()
+					if !ok {
+						sizeInBytes, ok = size.AsDec().Unscaled()
+						if !ok {
+							return nil, fmt.Errorf("failed to parse volume size %s", dataVolume.Size)
+						}
+					}
+					w["kubeletDataVolume"] = map[string]interface{}{
+						"type": "test",
+						"size": fmt.Sprintf("%d", sizeInBytes),
```

Comment:
Can't you use 

```suggestion
						"size": sizeInBytes,
```

directly?

Reply:
it would show the scientific (exponential) notation of the number, and cause a mismatch in the mount script that relies on the format of the `lsblk`